subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/EclipsePlugin.java

```diff
@@ -208,58 +237,65 @@ public void execute(final GenerateEclipseClasspath task) {
 
                         //model properties:
                         task.setClasspath(model.getClasspath());
-                        task.getClasspath().setFile(new XmlFileContentMerger(task.getXmlTransformer()));
-                        task.getClasspath().setSourceSets(project.getConvention().getPlugin(JavaPluginConvention.class).getSourceSets());
-
-                        AfterEvaluateHelper.afterEvaluateOrExecute(project, new Action<Project>() {
-                            @Override
-                            public void execute(Project p) {
-                                // keep the ordering we had in earlier gradle versions
-                                Set<String> containers = Sets.newLinkedHashSet();
-                                containers.add("org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/" + model.getJdt().getJavaRuntimeName() + "/");
-                                containers.addAll(task.getClasspath().getContainers());
-                                task.getClasspath().setContainers(containers);
-                            }
-
-                        });
-
-                        project.getPlugins().withType(JavaPlugin.class, new Action<JavaPlugin>() {
-                            @Override
-                            public void execute(JavaPlugin javaPlugin) {
-                                configureJavaClasspath(project, task);
-                            }
-
-                        });
+                    }
+                });
 
-                        configureScalaDependencies(project, task);
+                model.getClasspath().setFile(ValueCachingProvider.of(task.map(new Transformer<XmlFileContentMerger, GenerateEclipseClasspath>() {
```

Comment:
We should move the Xml transformer to the model and use it from the task instead.